open-metadata-implementation/adapters/open-connectors/data-store-connectors/cassandra-data-store-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/datastore/cassandra/CassandraDataStoreConnector.java

```diff
@@ -0,0 +1,157 @@
+import com.datastax.oss.driver.api.core.CqlSessionBuilder;
+import com.datastax.oss.driver.api.core.metadata.schema.SchemaChangeListener;
+import org.odpi.openmetadata.adapters.connectors.datastore.cassandra.ffdc.CassandraDataStoreAuditCode;
+import org.odpi.openmetadata.frameworks.connectors.ConnectorBase;
+import org.odpi.openmetadata.frameworks.connectors.ffdc.ConnectorCheckedException;
+import org.odpi.openmetadata.frameworks.connectors.properties.ConnectionProperties;
+import org.odpi.openmetadata.frameworks.connectors.properties.EndpointProperties;
+import org.odpi.openmetadata.repositoryservices.auditlog.OMRSAuditLog;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetSocketAddress;
+
+/**
+ * The Cassandra data store connector provides the OCF connection objects for Apache Cassandra Database.
+ */
+public class CassandraDataStoreConnector extends ConnectorBase {
+    private static final Logger log = LoggerFactory.getLogger(CassandraDataStoreConnector.class);
+    private CassandraDataStoreAuditCode auditLog;
+    private OMRSAuditLog omrsAuditLog;
+
+    private CqlSession cqlSession;
+    private String username = null;
+    private String password = null;
+    private String serverAddresses = null;
+    private Integer port = 9042;
+
+    /**
+     * Typical Constructor: Connectors should always have a constructor requiring no parameters and perform
+     * initialization in the initialize method.
+     */
+    public CassandraDataStoreConnector() {
+        super();
+    }
+
+    /**
+     * Call made by the ConnectorProvider to initialize the Connector with the base services.
+     *
+     * @param connectorInstanceId  unique id for the connector instance   useful for messages etc
+     * @param connectionProperties POJO for the configuration used to create the connector.
+     */
+    @Override
+    public void initialize(String connectorInstanceId, ConnectionProperties connectionProperties) {
+        super.initialize(connectorInstanceId, connectionProperties);
+
+        final String actionDescription = "initialize Cassandra Data Store Connection";
+
+        this.connectorInstanceId = connectorInstanceId;
+        this.connectionProperties = connectionProperties;
+
+        EndpointProperties endpoint = connectionProperties.getEndpoint();
+
+        super.initialize(connectorInstanceId, connectionProperties);
+
+        if (omrsAuditLog != null) {
+            auditLog = CassandraDataStoreAuditCode.CONNECTOR_INITIALIZING;
+            omrsAuditLog.logRecord(
+                    actionDescription,
+                    auditLog.getLogMessageId(),
+                    auditLog.getSeverity(),
+                    auditLog.getFormattedLogMessage(),
+                    null,
+                    auditLog.getSystemAction(),
+                    auditLog.getUserAction());
+        }
+
+        if (endpoint != null ) {
+            serverAddresses = endpoint.getAddress();
```

Comment:
I think endpoint address contains both hostname and port,  it can be used instead of the default port.

Reply:
Yes I have added it with additionalProperty